interface/main/messages/messages.php

```diff
@@ -23,614 +24,686 @@
+                    text($name) . "</td><td width=5></td></tr></table></td>
+        <td style=\"border-bottom: 1px #000000 solid; border-right: 1px #000000 solid;\"><table cellspacing=0 cellpadding=0 width=100%><tr><td width=5></td><td class=\"text\"><a href=\"messages.php?showall=".attr($showall)."&sortby=".attr($sortby)."&sortorder=".attr($sortorder)."&begin=".attr($begin)."&task=edit&noteid=" .
+                    attr($myrow['id']) . "&$activity_string_html\" onclick=\"top.restoreSession()\">" .
+                    text($patient) . "</a></td><td width=5></td></tr></table></td>
+        <td style=\"border-bottom: 1px #000000 solid; border-right: 1px #000000 solid;\"><table cellspacing=0 cellpadding=0 width=100%><tr><td width=5></td><td class=\"text\">" .
+                    text($myrow['title']) . "</td><td width=5></td></tr></table></td>
+        <td style=\"border-bottom: 1px #000000 solid; border-right: 1px #000000 solid;\"><table cellspacing=0 cellpadding=0 width=100%><tr><td width=5></td><td class=\"text\">" .
+                    text( oeFormatShortDate(substr($myrow['date'], 0, strpos($myrow['date'], " "))) ) . "</td><td width=5></td></tr></table></td>
+        <td style=\"border-bottom: 1px #000000 solid;\"><table cellspacing=0 cellpadding=0 width=100%><tr><td width=5></td><td class=\"text\">" .
+                    text( $myrow['message_status'] ) . "</td><td width=5></td></tr></table></td>
         </tr>";
-        // Display the Messages table body.
-        $count = 0;
-        $result = getPnotesByUser($active,$show_all,$_SESSION['authUser'],false,$sortby,$sortorder,$begin,$listnumber);
-    while ($myrow = sqlFetchArray($result)) {
-        $name = $myrow['user'];
-        $name = $myrow['users_lname'];
-        if ($myrow['users_fname']) {
-            $name .= ", " . $myrow['users_fname'];
+                }
+                // Display the Messages table footer.
+                echo "
+      </form></table>
+      <table border=0 cellpadding=5 cellspacing=0 width=90%>
+      <tr>
+      <td class=\"text\"><a href=\"messages.php?showall=".attr($showall)."&sortby=".attr($sortby)."&sortorder=".attr($sortorder)."&begin=".attr($begin)."&task=addnew&$activity_string_html\" onclick=\"top.restoreSession()\">" .
+                xlt('Add New') . "</a> &nbsp; <a href=\"javascript:confirmDeleteSelected()\" onclick=\"top.restoreSession()\">" .
+                xlt('Delete') . "</a></td>
+      <td align=right class=\"text amount-msg\">$prevlink &nbsp; $end of $total &nbsp; $nextlink</td>
+      </tr>
+      </table></td></tr></table><br>";
+                ?>
+              <script language="javascript">
+                // This is to confirm delete action.
+                function confirmDeleteSelected() {
+                  if(confirm("<?php echo xlt('Do you really want to delete the selection?'); ?>")) {
```

Comment:
use xls() instead of xlt().
(this is because this is a javascript literal and is best to use addslashes() for that which is what xls() uses)